File: BotsSDK/korebotsdklib/src/main/java/kore/botssdk/models/StaticLinksModel.java
package kore.botssdk.models;

import java.io.Serializable;
import java.util.ArrayList;

import kore.botssdk.utils.StringUtils;

public class StaticLinksModel implements Serializable {
    Boolean show;
    String type;
    String layout;
    private ArrayList<BrandingQuickStartButtonButtonsModel> links;

    public StaticLinksModel updateWith(StaticLinksModel configModel) {
        show = configModel.show != null ? configModel.show : show;
        type = !StringUtils.isNullOrEmpty(configModel.type) ? configModel.type : type;
        layout = !StringUtils.isNullOrEmpty(configModel.layout) ? configModel.layout : layout;
        links = configModel.links != null && !configModel.links.isEmpty() ? configModel.links : links;
        return this;
    }

    public void setShow(boolean show) {
        this.show = show;
    }

    public void setLinks(ArrayList<BrandingQuickStartButtonButtonsModel> links) {
        this.links = links;
    }

    public void setType(String type) {
        this.type = type;
    }

    public boolean isShow() {
        return show != null ? show : false;
    }

    public String getType() {
        return type;
    }

    public String getLayout() {
        return layout;
    }

    public ArrayList<BrandingQuickStartButtonButtonsModel> getLinks() {
        return links;
    }
}

File: BotSdk/botclient/src/main/java/com/kore/network/api/responsemodels/branding/StaticLinksModel.kt
package com.kore.network.api.responsemodels.branding

import com.google.gson.annotations.SerializedName

data class StaticLinksModel(
    @SerializedName("show")
    val show: Boolean?,
    @SerializedName("type")
    val type: String? = null,
    @SerializedName("layout")
    val layout: String = "",
    @SerializedName("links")
    val links: ArrayList<BrandingQuickStartButtonButtonsModel> = ArrayList()
)

fun StaticLinksModel.updateWith(configModel: StaticLinksModel?): StaticLinksModel {
    return this.copy(
        show = configModel?.show ?: show,
        type = configModel?.type?.ifEmpty { type } ?: type,
        layout = configModel?.layout?.ifEmpty { layout } ?: layout,
    )
}